infer/src/dotnet/ResourceLeakCSDomain.ml

```diff
@@ -73,53 +74,64 @@ let check_count access_path held =
   let old_count = find_count access_path held in
   match old_count with NonTop count when count > 0 -> true | _ -> false
 
+let reset_type_map = type_map := ResourcesTypeHeld.empty
+
+(** Return leaked types as string*)
+let type_to_list held = 
+  let leaked_type_list = ref [] in
+  let concat_types = 
+    ResourcesTypeHeld.iter
+      (fun x y ->
+        if check_count x held then
+          leaked_type_list := LeakList.append_one !leaked_type_list y )
```

Comment:
Should we worry about the quadratic complexity here? `append_one` appends at the end of `leaked_type_list`, which is linear, and we do that once for each binding in `type_map`, not sure how big that is. Usually it's better to build the list in reverse and reverse it at the end (assuming you care about the order) to do the whole thing in linear time.

Reply:
Done